Fix/grouping

zoemcl opened the pull request:
Implemented a ChangeGroup action in the editor toolkit that allows users to change the grouping of a neume upon click select.

zoemcl committed:
implemented changeGroup action in the editor toolkit.

zoemcl committed:
Updated m_info to be the Id of the new neume grouping

zoemcl committed:
Merge with develop

JRegimbal commented on src/editortoolkit.cpp:
```diff
@@ -924,9 +932,101 @@ bool EditorToolkit::Ungroup(std::string groupType, std::vector<std::string> elem
+        LogError("Could not get the drawing page.");
+        return false;
+    }
+    Neume *el = dynamic_cast<Neume *> (m_doc->GetDrawingPage()->FindChildByUuid(elementId));
```
Check to make sure this is not null. If the element does not exist then `el` will be set to `nullptr` and will cause a segmentation fault when trying to dereference it in a few lines.

zoemcl replied:
fixed in fbe8061

JRegimbal commented on src/editortoolkit.cpp:
```diff
@@ -924,9 +932,101 @@ bool EditorToolkit::Ungroup(std::string groupType, std::vector<std::string> elem
+        prevNc = newNc;
+
+    }
+    m_editInfo = firstChild->GetUuid();
```
The UUID should probably be that of the neume, not the first neume component. I don't see how that would be as useful.

zoemcl replied:
fixed in fbe8061

JRegimbal reviewed (changes requested):
One possible bug that should be fixed before merging. Otherwise seems good.

zoemcl committed:
Applied review comments

Added a check to make sure el is not a nullptr and changed the id given to m_info of that of el.
